crates/dojo-lang/src/inline_macros/set.rs

```diff
@@ -73,7 +99,56 @@ impl InlineMacroExprPlugin for SetMacro {
             };
         }
 
-        for entity in bundle {
+        let mut module_name = "".to_string();
+        let module_syntax_node =
+            get_parent_of_kind(db, &syntax.as_syntax_node(), SyntaxKind::ItemModule);
+        if let Some(module_syntax_node) = &module_syntax_node {
+            let mod_ast = ItemModule::from_syntax_node(db, module_syntax_node.clone());
+            module_name = mod_ast.name(db).as_syntax_node().get_text_without_trivia(db);
+        }
+
+        let mut fn_name = "".to_string();
+        let fn_syntax_node =
+            get_parent_of_kind(db, &syntax.as_syntax_node(), SyntaxKind::FunctionWithBody);
+        if let Some(fn_syntax_node) = &fn_syntax_node {
+            let fn_ast = FunctionWithBody::from_syntax_node(db, fn_syntax_node.clone());
+            fn_name = fn_ast.declaration(db).name(db).as_syntax_node().get_text_without_trivia(db);
+        }
+
+        for (entity, syntax_node) in bundle {
+            // db.lookup_intern_file(key0);
+            if !module_name.is_empty() && !fn_name.is_empty() {
```

Comment:
```suggestion
            if !module_name.is_empty() && !fn_name.is_empty() {
```

What is the case where we hit this? Is it an expected behavior that both of these won't be defined?

Reply:
We need both of these defined. It's unlikely that they are not. But just added the if statement for sanity.
Both of them have `!` so they are _not_ empty...

Reply:
In what case would they not both be defined? If it is unexpected, we should present an error rather than handling it silently

Reply:
Error messages added ;)
I think it shouldn't be a diagnostic error that stop the compilation, just a normal error log kinda thing. What do you think?

Reply:
Yes I think so. I still don't understand when this can happen. It would be useful to create an example test with the expected diagnostic output in the test data

Reply:
I don't think it will ever happen tbh.

Reply:
If we silently fail, it will be hard for us to figure out that a legal node structure we don't handle even exists. If we think it shouldn't happen, then we should fail and output an error. So the user will come to us and let us know there is something we don't handle yet. Otherwise it becomes a very hard to debug issue

Reply:
It seems to happen in tests, So I think it would happen for `get!`/`set!` in a function put directly in `lib.cairo`
<img width="359" alt="image" src="https://github.com/dojoengine/dojo/assets/11048263/08d59260-e054-4bbc-8f09-4aaec00d9230">